File: src/main/java/me/kuku/yuq/service/impl/QQServiceImpl.java
package me.kuku.yuq.service.impl;

import com.icecreamqaq.yudb.jpa.annotation.Transactional;
import me.kuku.yuq.dao.GroupDao;
import me.kuku.yuq.dao.QQDao;
import me.kuku.yuq.entity.GroupEntity;
import me.kuku.yuq.entity.QQEntity;
import me.kuku.yuq.service.QQService;

import javax.inject.Inject;
import java.util.List;

public class QQServiceImpl implements QQService {
    @Inject
    private QQDao qqDao;
    @Inject
    private GroupDao groupDao;
    @Override
    @Transactional
    public QQEntity findByQQAndGroup(Long qq, Long group) {
        GroupEntity groupEntity = groupDao.findByGroup(group);
        if (groupEntity == null) return null;
        return qqDao.findByQQAndGroup(qq, groupEntity.getId());
    }

    @Override
    @Transactional
    public void save(QQEntity qqEntity) {
        qqDao.saveOrUpdate(qqEntity);
    }

    @Override
    @Transactional
    public int delByQQAndGroup(Long qq, Long group) {
        GroupEntity groupEntity = groupDao.findByGroup(group);
        if (groupEntity == null) return 0;
        return qqDao.delByQQAndGroup(qq, groupEntity.getId());
    }

    @Override
    @Transactional
    public List<QQEntity> findAll() {
        return qqDao.findAll();
    }

    @Override
    @Transactional
    public List<QQEntity> findByHostLocPush(boolean hostLocPush) {
        return qqDao.findByHostLocPush(hostLocPush);
    }
}

File: src/main/kotlin/me/kuku/yuq/service/impl/QQServiceImpl.kt
package me.kuku.yuq.service.impl

import com.icecreamqaq.yudb.jpa.annotation.Transactional
import me.kuku.yuq.dao.GroupDao
import me.kuku.yuq.dao.QQDao
import me.kuku.yuq.entity.QQEntity
import me.kuku.yuq.service.QQService
import javax.inject.Inject

class QQServiceImpl: QQService {
    @Inject
    private lateinit var qqDao: QQDao
    @Inject
    private lateinit var groupDao: GroupDao

    @Transactional
    override fun findByQQAndGroup(qq: Long, group: Long): QQEntity? {
        val groupEntity = groupDao.findByGroup(group) ?: return null
        return qqDao.findByQQAndGroup(qq, groupEntity.id!!)
    }

    @Transactional
    override fun save(QQEntity: QQEntity) = qqDao.saveOrUpdate(QQEntity)

    @Transactional
    override fun delByQQAndGroup(qq: Long, group: Long): Int {
        val groupEntity = groupDao.findByGroup(group) ?: return 0
        return qqDao.delByQQAndGroup(qq, groupEntity.id!!)
    }

    @Transactional
    override fun findAll() = qqDao.findAll()
}